app/models/secure_data.rb

```diff
@@ -0,0 +1,22 @@
+# Note: model uses UUID for id which is generated by Postgres. Note that any changes to
+# the way the UUID is generated need to ensure that it generated by a secure random system.
+class SecureData < ApplicationRecord
+  def self.create_and_store(hash)
+    secure_data = new
+    secure_data.store(hash)
+    secure_data.save
+    secure_data.id
+  end
+
+  def self.for(id)
+    find(id).retrieve
+  end
+
+  def store(hash)
+    self.data = JWT.encode hash, nil, 'none'
+  end
+
+  def retrieve
+    JWT.decode(data, nil, false).first.symbolize_keys
```

Comment:
📝 Memo for signing the JWT (as spoken offline), as without it we can leave ourselves vulnerable by dynamically picking the algorithm, [as stated in the gem itself](https://github.com/jwt/ruby-jwt#algorithms-and-usage).